feat: 소셜 로그인을 구현한다

drunkenhw opened the pull request:
## 📄 Summary
>  소셜 로그인을 구현했습니다. 
만약 새로운 로그인 플랫폼을 추가한다면 
1. yml 파일에 해당 정보를 추가합니다
2. `OAuthMember`를 구현한 클래스를 생성합니다
3. `Provider` enum에 값을 추가합니다.
이런 flow로 새로운 플랫폼을 추가할 수 있습니다.

## 🙋🏻 More
> 


close #87

drunkenhw committed:
chore: application 의 property 읽는 의존성 추가

drunkenhw committed:
chore: application 의 property 읽는 의존성 추가

drunkenhw committed:
feat: OAuth2 회원 추가

drunkenhw committed:
feat: OAuth2 회원 추가

drunkenhw committed:
feat: provider 에게 회원 정보를 요청하는 기능 추가

drunkenhw committed:
feat: provider 에게 회원 정보를 요청하는 기능 추가

drunkenhw committed:
feat: 소셜 회원 정보를 조회하고 저장하는 기능 추가

drunkenhw committed:
feat: 소셜 회원 정보를 조회하고 저장하는 기능 추가

feb-dain committed:
feat: 버튼 스타일 컴포넌트를 구현한다 (#109)

* feat: 버튼 스타일 컴포넌트 구현

[#107]

* test: 설명 및 옵션 추가

[#107]

* feat: borderRadius에 `all` 옵션 추가

[#107]

-

Co-authored-by: Gabriel Ju Hyun, Yoon <gabrielyoon7@gmail.com>

kyw0716 committed:
refactor: google map을 전달하는 구조를 개선한다 (#111)

* refactor: 구글 지도 최초 로드 방식 수정

[#110]

* refactor: 구글 지도 사용 방식 수정

- 모든 컴포넌트에서 전역 상태에서 받아서 사용하는 방식으로 통일

[#110]

* refactor: 혼동될 수 있는 네이밍 수정

[#110]

* refactor: 혼동될 수 있는 네이밍 수정

[#110]

sosow0212 commented on backend/src/main/java/com/carffeine/carffeine/domain/member/MemberRepository.java:
```diff
@@ -0,0 +1,13 @@
+    Optional<Member> findByEmail(String email);
+
+    Member save(Member member);
+
```
불필요한 개행 삭제해주세요~

sosow0212 commented on backend/src/main/java/com/carffeine/carffeine/domain/member/exception/MemberExceptionType.java:
```diff
@@ -0,0 +1,34 @@
+import com.carffeine.carffeine.common.exception.Status;
+
+public enum MemberExceptionType implements ExceptionType {
+    INVALID_AUTH_PROVIDER(Status.UNAUTHORIZED, 2001, "지원하지 않는 로그인 플랫폼입니다"),
```
개행 한 칸 추가해주세요!

sosow0212 commented on backend/src/main/java/com/carffeine/carffeine/domain/member/oauth/OAuthMember.java:
```diff
@@ -0,0 +1,12 @@
+
+public interface OAuthMember {
+
+    String email();
```
잘 몰라서 그러는데 메서드명이 고정되어 있는건가요??
동사로 시작하면 더 좋을 것 같아서요!

sosow0212 commented on backend/src/main/java/com/carffeine/carffeine/domain/member/oauth/Provider.java:
```diff
@@ -0,0 +1,33 @@
+
+public enum Provider {
+
+    GOOGLE("google", GoogleMember::new),
```
enum으로 관리하면서 값을 가져오는 건가요? 나중에 다른 oauth가 추가되어도 좋을 것 같네요  👍

sosow0212 commented on backend/src/main/java/com/carffeine/carffeine/infra/oauth/RestTemplateOAuthRequester.java:
```diff
@@ -0,0 +1,77 @@
+    private final RestTemplate restTemplate;
+    private final OAuthProviderProperties oAuthProviderProperties;
+
+    public OAuthMember login(OAuthLoginRequest request) {
```
혹시 @Override 키워드를 제거하신 이유가 있나요?

sosow0212 commented on backend/src/test/java/com/carffeine/carffeine/infra/oauth/RestTemplateOAuthRequesterTest.java:
```diff
@@ -0,0 +1,44 @@
+
+@DisplayNameGeneration(DisplayNameGenerator.ReplaceUnderscores.class)
+@SuppressWarnings("NonAsciiCharacters")
+@SpringBootTest(webEnvironment = SpringBootTest.WebEnvironment.DEFINED_PORT)
```
잘 몰라서 그러는데 혹시 DEFINED_PORT로 통합테스트하신 이유가 있으신가요?

sosow0212 commented on backend/src/test/java/com/carffeine/carffeine/service/auth/AuthServiceTest.java:
```diff
@@ -0,0 +1,49 @@
+        assertThat(member).usingRecursiveComparison()
+                .isEqualTo(findMember);
+    }
+
```
불필요한 개행 삭제해주세요~

sosow0212 commented on backend/src/test/java/com/carffeine/carffeine/service/auth/OAuthProviderPropertiesTest.java:
```diff
@@ -0,0 +1,35 @@
+            softly.assertThat(oAuthProviderProperty.getRedirectUrl()).isEqualTo("google-redirect-url");
+        });
+    }
+
```
여기도 불필요한 개행 삭제해주세요!

sosow0212 commented on backend/src/test/java/com/carffeine/carffeine/service/auth/AuthServiceTest.java:
```diff
@@ -0,0 +1,49 @@
+    }
+
+    @Test
+    void 소셜_로그인_후_회원을_반환한다() {
```
로그인 정보가 없다면 Repository를 이용해서 저장을 하는 로직이 있는데, 그것도 테스트 하면 더 좋을 것 같아요~

sosow0212 reviewed (changes requested):
작업하시느라 고생하셨습니다~
몇 가지의 컨벤션 관련된 리뷰와 코멘트 남겨놨습니다 👍

feb-dain committed:
test: 필요없는 파일을 제거하고 스토리북 기본 형식을 변경한다 (#117)

* test: 필요없는 파일(스토리북 관련) 제거

[#115]

* test: 스토리북 기본 형식 변경

[#115]

* test: 필요없는 코드 제거

[#115]

gabrielyoon7 committed:
feat: 현위치 버튼 클릭 시 현위치로 복귀하는 기능을 구현한다 (#118)

* feat: 아무 기능이 없는 현위치 복귀 버튼을 추가

* feat: 현위치 버튼을 누르면 현 위치로 이동하는 기능을 구현

drunkenhw committed:
style: 컨벤션에 맞게 개행 추가 및 제거

[#87]

kyw0716 committed:
chore: 마커 커스텀을 위한 svg import 환경을 설정한다 (#122)

be-student commented on backend/src/test/java/com/carffeine/carffeine/infra/oauth/FakeGoogleController.java:
```diff
@@ -0,0 +1,62 @@
+@RestController
+public class FakeGoogleController {
+
+    @PostMapping("/v4/token")
```
저희 버저닝 하는 것을 안 정했는데, v4 를 붙힌 기준은 무엇인가요?

drunkenhw replied:
구글에서 이렇게 받는데요 제가 지정한 부분이 아니라 구글 api 주소라서 그렇습니다!

feb-dain committed:
chore: 커밋시 프리티어와 eslint 적용을 자동화한다 (#126)

* test: 필요없는 파일 및 코드 삭제

[#114]

* chore: husky, lint-staged 설치 및 적용

[#103]

* chore: eslint fix 자동화

[#103]

be-student commented on backend/src/main/resources/application.yml:
```diff
@@ -38,3 +36,12 @@ api:
+    google:
+      id: 482896035816-ldevmjo314h6d42vns69v7q58d7me3ks.apps.googleusercontent.com
+      secret: ENC(SqpABZuVRJAtZUv7L1/oCS4LzsaSduagft+YLUKOUM2Zr+tuXYg+/DqPMxd/vKd8)
+      redirect-url: http://localhost:8080/login/oauth2/code/google
```
redirect uri 가 한 곳으로 고정되어있으면 추후에 테스트하기 어려울 것 같아요

ex) prod 서버에서 redirect 를 시킬 때, 우리 도메인/login~~~ 일 수도 있고
localhost/login 일 수도 있으니까요
유연하게 해주면 좋을 것 같아요 

이 부분만 더 유연하게 해주면 좋을 것 같아요

be-student reviewed (changes requested):
화이팅!
다른 부분은 다 잘 작성해주셨네요 👍

kiarakim commented on backend/src/main/java/com/carffeine/carffeine/service/auth/OAuthProviderProperties.java:
```diff
@@ -0,0 +1,32 @@
+        return this.provider.get(provider);
+    }
+
+    @Getter
```
클래스의 
@Getter랑 이 메서드의 @Getter는 다른가요?

kyw0716 committed:
feat: 충전소 마커를 필터링 하는 기능을 구현한다 (#124)

* fix: SetStateCallback 타입 추가 반영

[#123]

* test: 사용 가능한 충전기 개수 0개도 포함하도록 수정

[#123]

* feat: 충전소 필터링 옵션 전역 상태 추가

[#123]

* feat: 필터링 옵션 선택 컴포넌트 추가

[#123]

* feat: 충전소 정보 필터링 기능 추가

[#123]

* refactor: 직관적인 네이밍 적용

[#123]

* refactor: 직관적인 네이밍 적용

[#123]

kiarakim commented on backend/src/test/java/com/carffeine/carffeine/domain/member/oauth/ProviderTest.java:
```diff
@@ -0,0 +1,37 @@
+    }
+
+    @Test
+    void OAuth_로그인_API_제공자의_이름이_없으면_예외가_발생한다() {
+        // given
+        String wrongName = "gooooooogle";
+
+        // when & then
+        assertThatThrownBy(() -> Provider.from(wrongName))
+                .isInstanceOf(MemberException.class)
+                .hasMessage("지원하지 않는 로그인 플랫폼입니다");
+    }
+}
```
대소문자 틀린 이름을 검증하는 테스트도 하나 넣어주세염

kiarakim reviewed (changes requested):
Getter는 질문이고 test는 request changes 입니당

feb-dain committed:
fix: husky에 `commit-msg` 추가 (#133)

[#131]

kyw0716 committed:
feat: 간단 정보 띄우는 기능 구현 (#132)

[#125]

gabrielyoon7 committed:
feat: 마지막 접속한 이력이 있는 클라이언트라면 마지막 위치를 우선하여 지도를 로드하는 기능을 추가한다 (#134)

* feat: 마지막 접속한 이력이 있는 클라이언트라면 마지막 위치를 우선하여 지도를 로드하는 기능 추가

[#130]

* refactor: 불필요한 import 제거

[#130]

feb-dain committed:
fix: mac 전용 prepare script 추가 (#137)

[#136]

kyw0716 committed:
fix: 웹팩 빌드 오류를 해결한다 (#138)

* chore: 웹팩 빌드 오류 수정

- styled-components의 createGlobalStyle 내부에 @import 삭제

[#135]

* refactor: 개발 환경에서만 msw가 동작하도록 분기처리한 부분 삭제

- 백엔드와 연결 성공 시 다시 분기 처리할 것

[#135]

drunkenhw committed:
fix: 위도, 경도 기반으로 조회 시 위도가 잘못되는 버그를 해결한다  (#141)

* fix: 계산 로직을 수정한다

[#139]

* refactor: entity graph를 활용해서 발생하는 쿼리를 줄인다

[#139]

kiarakim committed:
feat: 충전기 상태 및 혼잡도를 저장하는 기능을 구현한다 (#113)

* feat: 충전기 상태 업데이트 기능

[#21]

* feat: 주기적으로 충전기 정보 업데이트 기능 구현

[#21]

* feat: 혼잡도 정보 업데이트 기능 구현

[#21]

* refactor: 사용하지 않는 메서드 제거 및 매직넘버 처리

* refactor: 코드리뷰를 반영한다

[#21]

* feat: 혼잡도를 조회하는 컨트롤러 작성

[#21]

[#21)]

* feat: adoc 작성

[#21]

* feat: 처음 저장시 status 도 함께 저장하도록 변경

[#21]

* fix: charger 조회시, status null 인 경우 false를 반환하도록 한다

[#21]

---------

Co-authored-by: be-student <presentsong@naver.com>

gabrielyoon7 committed:
fix: 몇몇 패키지를 devDenpencies에서 dependencies으로 이동한다 (#147)

* chore: 불필요한 파일 삭제

* chore: react-query 이전

* chore: webpack 이전

* chore: webpack-cli 이전

* chore: html-webpack-plugin 이전

* chore: copy-webpack-plugin 이전

* chore: dotenv-webpack 이전

* chore: ts-loader 이전

* chore: @types/google.maps 이전

* chore: @types/react @types/react-dom 이전

* chore: styled-components @types/styled-components 이전

* chore: msw 이전

* chore: @tanstack/react-query-devtools 이전

* chore: file-loader 이전

* chore: @storybook/react 이전

gabrielyoon7 committed:
feat: App을 import 하기 전에 구글 맵 키를 반드시 입력해야 실행이 되도록 하는 기능 구현 (#150)

feb-dain committed:
feat: 충전소 상세 정보 보여주는 기능을 구현한다 (#155)

* refactor: type 추가 및 변경

- 간단한 충전소 정보 타입과 자세한 충전소 정보 타입 분리
- mock 데이터용 타입 생성

[#128]

* test: 충전소 상세 정보 가져오는 mocking api 구현

[#128]

* feat: 선택된 충전소 id를 전역으로 관리하는 store 구현

[#128]

* feat: 선택된 충전소의 상세 정보를 가져오는 기능 구현

[#128]

* feat: 선택된 충전소의 상세 정보를 보여주는 컴포넌트 구현

[#128]

* feat: 충전소 목록에서 충전소 클릭시 해당 충전소의 상세 정보를 보여주는 기능 구현

[#128]

* feat: 간단 정보 모달에서 상세보기 버튼 클릭시 상세 정보를 보여주는 기능 구현

[#128]

* feat: 충전소 상세 정보 변경을 도와주는 커스텀 훅 구현

[#128]

* refactor: 충전소 세부 정보 fetch할 때 에러 처리하도록 변경

[#128]

* feat: 충전소 선택할 때마다 선택된 충전소의 상세 정보를 보여주는 기능 구현

[#128]

* refactor: 더 정확한 타입으로 변경

[#128]

* refactor: 충전소 간단 정보 요청 방식을 `POST` → `GET`으로 변경

[#128]

* refactor: 충전소 상세 정보 더 구체적으로 변경

[#128]

* refactor: api 타입 변경

[#128]

sosow0212 committed:
feat: Github Actions Self Hosted Runner를 위해 workflow 파일을 생성한다 (#127)

* refactor: 중복 코드 제거

* feat: cd 스크립트 작성

* refactor: DB 사용하도록 CD 스크립트 변경

* feat: 환경별로 profile 추가

* refactor: local profile을 사용하도록 변경

* refactor: Dockerfile Dev프로필 사용하도록 설정

* refactor: 옵션은 코드 레벨에서 관리할 수 있도록 리팩토링 진행

* refactor: 불필요한 동작 제거

* refactor: db 정보 암호화

* refactor: cd 스크립트 수정

sosow0212 committed:
feat: 충전기와 충전소를 주기적으로 업데이트하는 기능을 구현한다 (#151)

* feat: 업데이트 됐는지 확인하는 기능 추가

Co-authored-by: drunkenhw <drunkenhw@gmail.com>

* test: 기능 테스트 추가

Co-authored-by: drunkenhw <drunkenhw@gmail.com>

* feat: 충전기 및 충전소 업데이트 기능 추가

Co-authored-by: drunkenhw <drunkenhw@gmail.com>

* test: 충전기 및 충전소 업데이트 기능 테스트

Co-authored-by: drunkenhw <drunkenhw@gmail.com>

* refactor: address 필드를 Charger가 아닌 ChargeStation이 가지도록 변경

Co-authored-by: drunkenhw <drunkenhw@gmail.com>

* refactor: 팀 컨벤션에 맞게 코드 컨벤션 수정

Co-authored-by: drunkenhw <drunkenhw@gmail.com>

* feat: 업데이트 됐는지 확인하는 기능 추가

Co-authored-by: drunkenhw <drunkenhw@gmail.com>

* test: 기능 테스트 추가

Co-authored-by: drunkenhw <drunkenhw@gmail.com>

* feat: 충전기 및 충전소 업데이트 기능 추가

Co-authored-by: drunkenhw <drunkenhw@gmail.com>

* test: 충전기 및 충전소 업데이트 기능 테스트

Co-authored-by: drunkenhw <drunkenhw@gmail.com>

* refactor: address 필드를 Charger가 아닌 ChargeStation이 가지도록 변경

Co-authored-by: drunkenhw <drunkenhw@gmail.com>

* refactor: 팀 컨벤션에 맞게 코드 컨벤션 수정

Co-authored-by: drunkenhw <drunkenhw@gmail.com>

* refactor: 충전소를 업데이트 기능 메서드 병합

[#26]

* feat: 업데이트 됐는지 확인하는 기능 추가

Co-authored-by: drunkenhw <drunkenhw@gmail.com>

* test: 기능 테스트 추가

Co-authored-by: drunkenhw <drunkenhw@gmail.com>

* feat: 충전기 및 충전소 업데이트 기능 추가

Co-authored-by: drunkenhw <drunkenhw@gmail.com>

* fix: 컨플릭트 해결

Co-authored-by: drunkenhw <drunkenhw@gmail.com>

* refactor: 필요한 기능 제외하고 제거

Co-authored-by: drunkenhw <drunkenhw@gmail.com>

* test: 업데이트 기능 테스트 추가

Co-authored-by: drunkenhw <drunkenhw@gmail.com>

* refactor: 안쓰는 클래스 제거

Co-authored-by: drunkenhw <drunkenhw@gmail.com>

* refactor: equals & hashcode 재정의

Co-authored-by: drunkenhw <drunkenhw@gmail.com>

---------

Co-authored-by: drunkenhw <drunkenhw@gmail.com>

kyw0716 committed:
refactor: 급속/완속 구별 로직을 수정한다 (#156)

* refactor: 간단 정보 급속/완속 구분 기준 수정

[#154]

* refactor: 급속/완속 구분 기준 수정 내용 필터링 기능에 반영

[#154]

* feat: 필터링 후 화면에 없는 충전소의 상세 정보를 지우는 기능 추가

[#154]

* fix: 급속 수정

[#154]

drunkenhw committed:
refactor: capacity가 없을 경우 기본값을 정한다 (#153)

* refactor: capacity가 없을 경우 기본값을 정한다

[#144]

* fix: rebase 오류 해결

[#144]

gabrielyoon7 committed:
refactor: 실제 api와 클라이언트를 연결해본다 (#159)

* refactor: 충전소 데이터 수신부를 실제 api 명세에 맞춰서 수정

- msw 도 함께 수정

Co-authored-by: 한우석 <106640954+drunkenhw@users.noreply.github.com>
Co-authored-by: Dain Lee <108778921+feb-dain@users.noreply.github.com>

* refactor: 개발모드에서만 msw가 실행되게 수정

---------

Co-authored-by: 한우석 <106640954+drunkenhw@users.noreply.github.com>
Co-authored-by: Dain Lee <108778921+feb-dain@users.noreply.github.com>

kiarakim committed:
fix: 암호화 방법과 디코딩 키 변경 및 flyway 제거 (#162)

* fix: 암호화 방법과 디코딩 키 변경 및 flyway 제거

* fix: /제거

kiarakim commented on backend/src/main/java/com/carffeine/carffeine/domain/member/exception/MemberExceptionType.java:
```diff
@@ -0,0 +1,34 @@
+    private final int exceptionCode;
+    private final String message;
+
+    MemberExceptionType(Status status, int exceptionCode, String message) {
+        this.status = status;
+        this.exceptionCode = exceptionCode;
+        this.message = message;
+    }
```
우리가 원래 갖고 있는 ChargeStationExceptionType이 있는데  MemberExceptionType 도 만들었네요. 그럼 나중에 다른 기능 있으면 예를 들어 "토큰 생성 및 검증" 기능이 있으면 그거에 대한 ExceptionType 구현 enum 만들면 될까욥

kiarakim commented on backend/src/main/java/com/carffeine/carffeine/infra/oauth/RestTemplateOAuthRequester.java:
```diff
@@ -0,0 +1,77 @@
+        return provider.getOAuthProvider(getUserAttributes(property, token));
+    }
+
+    private OAuthTokenResponse requestAccessToken(OAuthProviderProperty property, OAuthLoginRequest codeRequest) {
+        HttpHeaders headers = new HttpHeaders();
+        headers.setBasicAuth(property.getId(), property.getSecret());
+        headers.setContentType(MediaType.APPLICATION_FORM_URLENCODED);
+
+        HttpEntity<MultiValueMap<String, String>> request = new HttpEntity<>(headers);
+        URI tokenUri = getTokenUri(property, codeRequest.code());
+        return restTemplate.postForEntity(tokenUri, request, OAuthTokenResponse.class).getBody();
+    }
+
+    private URI getTokenUri(OAuthProviderProperty property, String code) {
+        return UriComponentsBuilder.fromUriString(property.getTokenUrl())
+                .queryParam(CODE, URLDecoder.decode(code, StandardCharsets.UTF_8))
+                .queryParam(GRANT_TYPE, AUTHORIZATION_CODE)
+                .queryParam(REDIRECT_URI, property.getRedirectUrl())
+                .build()
+                .toUri();
+    }
```
여기 Token은 무슨 토큰인가유. 대면 설명해주세요

kiarakim reviewed (approved):
수정할 건 없고 질문들입니다!
그 입력값의 대소문자 구분하는 테스트 부분만 고려해주십셔

feb-dain committed:
refactor: 서버 주소 추가 및 서버가 주는 정보에 맞춰 코드 변경한다  (#163)

* refactor: 잘못된 데이터가 들어왔을 때 null 값으로 변경

[#160]

* refactor: 서버 주소 추가

[#160]

* refactor: 개발 모드, 프로덕션 모드에 따라 다른 script 실행하도록 추가

[#160]

* refactor: 잘못된 값 리스트 상수화

[#160]

* refactor: 파라미터 불변성 지키는 방향으로 개선

[#160]

* refactor: 에러 메시지 상수화

[#160]

be-student committed:
feat: FE의 CD 를 구축한다 (#152)

* feat: fe cd 구축

* fix: 조건을 release/frontend 에 push 될 때로 변경한다

[#149]

gabrielyoon7 committed:
feat: 텍스트 스타일 컴포넌트를 제작한다  (#145)

* feat: Text 컴포넌트 구현

[#143]

* test: Text Component 스토리북 작성

[#143]

* refactor: 텍스트 컴포넌트 설명 수정

[#143]

* chore: 허스키 설정 수정

[#143]

* chore: font 설정

[#143]

* refactor: 텍스트 사이즈 수정

[#143]

* feat: 글자의 기본 하단 간격을 주는 기능 구현

[#143]

* test: marginBottom 스토리북 작성

[#143]

* test: marginBottom 스토리북 show cases 추가

[#143]

* test: 글자 정렬하는 스토리북 테스트 추가

[#143]

* fix: 잘못된 프로퍼티 키 값 수정

[#143]

* feat: body 사이즈 재조정

[#143]

* refactor: 주석 위치 수정

[#143]

* refactor: Text컴포넌트의 마진 개선

- marginBottom => mb
- 단위를 0.4rem으로 개선

[#143]

* refactor: Text storybook description 개선

[#143]

* refactor: 잘못된 변수명 개선

[#143]

* refactor: Text Storybook description 개선

[#143]

* refactor: Text Storybook 개선

[#143]

* refactor: 기본 폰트 사이즈 수정 및 타이틀 폰트 사이즈 수정

[#143]

* refactor: 타이틀 폰트 사이즈 수정

[#143]

drunkenhw committed:
refactor: template에 예상 소요 시간, 실제 소요 시간을 추가한다 (#166)

* refactor: PR template에 실제 작업 시간 추가

* refactor: issue template에 실제 작업 시간 추가

* docs: 소요 시간을 뜻하는 단어 변경

drunkenhw committed:
refactor: 패키지 분리 (#174)

Co-authored-by: sosow0212 <sosow0212@naver.com>
Co-authored-by: be-student <presentsong@naver.com>
Co-authored-by: kiarakim <kiarakim413@gmail.com>

[#171]

feb-dain committed:
feat: 플렉스 박스 스타일 컴포넌트를 구현한다 (#173)

* test: 필요없는 파일 및 코드 삭제

[#114]

* test: 타입 중복 제거

- 재사용되는 타입은 전역으로 분리
- 컴포넌트의 Props export하여 스토리북에서 재사용

[#114]

* feat: FlexBox 컴포넌트 구현

[#114]

* design: GlobalStyle에서 `font` 적용하는 선택자 변경

이유
- font가 적용 안 되는 상황 방지
- 스토리북 font도 Noto Sans KR로 변경

[#114]

* chore: storybook 접근성 검사 addon 설치

[#114]

* refactor: 공통으로 사용되는 스타일 전역으로 분리

[#114]

* fix: 공통 style 전역으로 분리하면서 생기는 에러 개선

절대 경로 설정

[#114]

* refactor: 타입 이름 명확하게 변경

[#114]

* refactor: `justify content`와 `align items` 속성 따로 적용할 수 있도록 변경

[#114]

* test: row gap과 column gap 설명 보강

[#114]

* refactor: px → rem 단위로 변경

[#114]

feb-dain committed:
refactor: FlexBox에 width, height 속성을 추가한다 (#178)

* refactor: FlexBox에 width, height 속성 추가

[#177]

* refactor: width/height를 적용하지 않았을 때, 어떤 값이 적용되는지 코드만 봐도 알 수 있도록 개선

[#177]

feb-dain committed:
fix: 스토리북 svg 경로 에러를 해결한다 (#184)

* refactor: px → rem으로 변경

[#175]

* fix: `@assets` 절대 경로 스토리북에 추가

[#183]

* chore: @svgr/webpack 설치

[#183]

* fix: storybook svg 경로 에러 개선

[#183]

kyw0716 committed:
fix: @style 경로 참조 에러 관련 웹팩 설정 추가 (#190)

[#189]

kyw0716 committed:
feat: 충전소 정보 서버 필터링 api를 mocking한다 (#188)

* feat: 서버 필터링 api mocking

[#169]

* refactor: 서버 필터링 부분 코드 가독성 개선

[#169]

* refactor: ,를 통해 여러 query param을 전달할 수 있도록 하는 유틸함수 추가

- URLSearchParams는 ,를 사용할 수 없어 새로운 유틸함수 추가함

[#169]

* feat: 간단한 예시 코드 추가

[#169]

* refactor: 실제 반환 데이터를 기반으로 mock data 생성하도록 수정

[#169]

* refactor: 충전기 타입, 회사명 상수화

[#169]

* feat: 충전소 세부정보에 상수화 된 충전기 타입 적용

[#169]

* refactor: 불필요한 console.log 삭제

[#169]

* refactor: 미리 상수화 해둔 객체를 활용해 mock data를 생성하도록 수정

[#169]

* refactor: url 변환 함수 코드 개선

[#169]

* fix: 잘못된 상수 수정

[#169]

kyw0716 committed:
feat: 네비게이터 컴포넌트를 구현한다 (#191)

* feat: Navigator 컴포넌트 추가

[#185]

* design: StationList 컴포넌트 위치 수정

[#185]

* feat: 지도에 Navigator 컴포넌트 추가

[#185]

* feat: Navigator에 아이콘 추가

[#185]

* design: cursor pointer 속성 추가

[#185]

* refactor: 로고 변경

[#185]

* refactor: px → rem

[#185]

kyw0716 committed:
feat: 실시간 필터링 옵션에 외부인 출입 제한 여부에 대한 선택 옵션 추가 (#193)

[#186]

gabrielyoon7 committed:
refactor: Text variant type 지정 (#194)

[#192]

drunkenhw committed:
feat: 충전소를 고장 신고하는 기능을 구현한다 (#196)

* feat: 고장 신고 클래스 추가

[#180]

* feat: 고장 신고 저장하는 기능 추가

[#180]

* feat: 고장 신고 기능 추가

[#180]

* feat: 고장 신고 api 추가

[#180]

* feat: 이미 신고한 충전소 검증 기능 추가

[#180]

gabrielyoon7 committed:
feat: List와 ListItem 컴포넌트를 구현한다 (#197)

* feat: List 컴포넌트 추가

[#172]

* feat: ListItem 컴포넌트 추가

[#172]

* fix: 오류로 인해 사라진 ListItem 컴포넌트 복구

[#172]

* feat: List Storybook 및 ListItem readme 추가

* feat: List에 디자인 옵션 추가

[#172]

* feat: ListItem에 clickable과 divider 추가

[#172]

* test: Menu 예제 추가

[#172]

* fix: Text 스토리타입 타입 오류 수정

[#172]

* test: List로 제어할 수 있는 예제 추가

[#172]

* feat: CSSProp 기능 추가

[#172]

* test: ListItem 스토리북 예제 구현

[#172]

* refactor: Text mb가 0일 때 아무것도 하지 않도록 개선

[#172]

gabrielyoon7 committed:
feat: 사용자 식별이 가능하도록 토큰을 생성하는 기능 구현 (#199)

* feat: 최초 접속 시 랜덤한 8자리 정수의 토큰을 발행하여 로컬 스토리지에 저장하는 기능 구현

- 단, 이미 발행된 토큰이 발견되는 경우에는 재발급 하지 않음

* refactor: 불필요한 콘솔 제거

* refactor: 스토리지 키 위치 수정

[#198]

feb-dain committed:
refactor: 절대 경로 확장성 있게 수정한다 (#205)

[#203]

feb-dain committed:
chore: `@heroicons/react`를 설치한다 (#202)

[#200]

feb-dain committed:
chore: `@heroicons/react`를 설치한다 (#202)

[#200]

gabrielyoon7 committed:
feat: develop mode에서 msw와 로컬 서버를 동적으로 선택하는 기능을 구현한다 (#215)

* refactor: msw를 실행하는 함수 분리

[#210]

* feat: msw를 키고 끄는 버튼 추가

[#210]

* feat: msw 컨트롤 버튼이 development 모드에서만 보이도록 수정

[#210]

* docs: 프론트엔드 설치 및 실행 방법 설명 추가

[#210]

* refactor: isMswMode로 상태명 변경

[#210]

kyw0716 committed:
feat: 서버 필터링 옵션을 선택하는 UI를 구현한다 (#207)

* refactor: FilterButtonList → StationFilterButtonList로 네이밍 변경

[#187]

* feat: 서버 필터링 기능 선택 UI 추가

[#187]

* feat: Navigator 컴포넌트에서 ServerStationFilters 컴포넌트 열고 닫는 기능 구현

[#187]

* refactor: StationFilterButtonList → ClientStationFilters로 네이밍 변경

[#187]

* refactor: 충전 속도 타입 추가

[#187]

* feat: 지도에 ServerStationFilter 컴포넌트 추가

[#187]

* design: 필터링 컴포넌트 우측 border 추가

[#187]

* feat: 서버 필터링 선택 UI 추가

[#187]

* refactor: FlexBox 컴포넌트 props 추가 → alignItems

[#187]

* fix: 잘못 들어간 타입 수정

[#187]

* style: props 넘길 때 '' → {''}로 수정

[#187]

* feat: 서버 필터링 옵션 적용하기 기능 구현

[#187]

* refactor: map을 참조하는 방식 수정

- 기존에 map을 인자로 넘겨 받아 수행하던 작업들 store에서 직접 참조하는 방식으로 변경

[#187]

* refactor: 필터링 버튼 사이의 간격 추가

[#187]

* design: 아이콘 크기 수정

[#187]

* refactor: 색상 표기 방식 변경 black → #000

[#187]

* refactor: 절대 경로 수정

[#187]

* design: 우아한 다크모드 삭제

[#187]

gabrielyoon7 committed:
feat: Box 컴포넌트를 구현한다 (#219)

* feat: Box 컴포넌트 추가

[#217]

* feat: border 추가

[#217]

* test: border 예제 추가

[#217]

* feat: p 기능 구현

[#217]

* feat: p 기능 완성

[#217]

* feat: m 기능 완성

[#217]

* feat: 높이, 너비 기능 완성

[#217]

* test: 높이, 너비 테스트

[#217]

* feat: bgColor, color 구현

[#217]

* test: bgColor, color 스토리북 작성

[#217]

* feat: position 관련 기능 추가

[#217]

* test: position 관련 기능 예제 추가

[#217]

* test: 부족한 description 추가

[#217]

* test: 부족한 description 추가

[#217]

* refactor: 리뷰 반영

[#217]

drunkenhw committed:
feat: 고장 신고 여부를 확인하는 기능을 구현한다  (#201)

* feat: 고장 신고를 회원과 충전소 기준으로 찾는 기능 추가

[#181]

* feat: 고장 신고를 회원과 충전소 기준으로 찾는 API 추가

[#181]

drunkenhw committed:
feat: 스토리북 배포를 자동화한다 (#206)

gabrielyoon7 committed:
feat: TextField를 구현한다 (#222)

* fix: 잘못된 속성 제거

[#220]

* feat: 기본 Input 컴포넌트 작성

[#220]

* feat: TextField로 변경 및 디자인 추가

[#220]

* refactor: 불필요한 컴포넌트 제거

[#220]

* feat: placeHolder 기능 추가 및 css 프롭 분리

[#220]

* feat: value, onChange 추가

[#220]

* feat: helperText 구현

[#220]

* test: description 개선

[#220]

* feat: width 기능 구현

[#220]

* refactor: 피드백 반영

[#220]

* refactor: 함수 순서 변경

[#220]

* refactor: 피드백 반영

[#220]

feb-dain committed:
feat: 검색창 컴포넌트를 구현한다 (#221)

* refactor: px → rem으로 변경

[#175]

* feat: StationSearch 컴포넌트 구현

[#175]

* refactor: getSize 함수 전역으로 관리

[#175]

* refactor: Button에 width, height 속성 추가

[#175]

* refactor: ListItem에 필요없는 속성 제거하고 NoLastDivider 속성 추가

[#175]

* refactor: gitSize 전역으로 관리

[#175]

* refactor: Text 타입 깔끔하게 분리

[#175]

* refactor: 검색 컴포넌트 구현

[#175]

* refactor: 검색바 분리 및 속성 추가

[#175]

* refactor: Text 스타일 컴포넌트에 새로운 속성 추가

[#175]

* refactor: 스타일 아래로 이동

[#175]

* refactor: 필요없는 코드 제거

[#175]

* feat: 검색 결과 리스트 컴포넌트 구현

[#175]

* feat: 충전소 검색창 컴포넌트 구현

[#175]

kyw0716 committed:
feat: 누적 혼잡도 통계를 시각화 해주는 컴포넌트를 개발한다 (#223)

* refactor: 불필요한 import 삭제

* refactor: 허스키 삭제

* refactor: 필터 적용하기 버튼 sticky 적용

* feat: 수치에 따라 길이가 변하는 그래프 바 구현

* feat: 혼잡도 통계 그래프 UI 구현

- 이전에 구현한 그래프 바 이용
- 헤더에 요일 버튼 UI 구현
- 그래프 바에 시간 정보 추가

* feat: 누적 혼잡도 ratio가 -1일 경우 회색으로 표시하는 기능 구현

* refactor: 구조분해 할당 적용

* feat: 급속/완속, 요일 변경 기능 추가

* feat: 선택된 요일 표시 기능 추가

* refactor: 급속/완속 토글 메서드 코드 개선

- if 분기처리 → 삼항연산자

feb-dain committed:
feat: pr시 comment에 스토리북 배포 링크를 자동으로 달아주는 기능을 구현한다 (#227)

[#226]

feb-dain committed:
feat: 검색창 렌더링 기능을 구현하고 웹 접근성 요소를 추가한다 (#231)

* refactor: 폴더 구조 개선

[#224]

* feat: 검색창 컴포넌트 여닫는 기능 구현

[#224]

* refactor: inline css 분리

[#224]

* feat: 검색창 이벤트 핸들러 임시로 구현(onChange, onSubmit)

[#224]

* test: 스토리북 title Components → UI로 변경

[#224]

* refactor: Navigator 폴더 생성 및 이동

[#224]

* refactor: Text  스타일 컴포넌트에 새로운 속성 추가

[#224]

* refactor: Navigator에 있는 svg 파일 아이콘으로 대체

[#224]

* design: StationSearchWindow 디자인 수정

[#224]

* refactor: Text 컴포넌트에 tag 바꾸는 속성 추가

[#224]

* refactor: 웹 접근성 개선

[#224]

* refactor: 시맨틱 태그로 변경

[#224]

* fix: 검색창을 열었을 때 필터를 클릭해도 필터창이 열리지 않는 오류 개선

[#224]

gabrielyoon7 committed:
feat: 모달을 구현한다 (#229)

* feat: 모달 열고 닫는 기능 구현

[#228]

* feat: 긴 모달 처리 구현

[#228]

* feat: 페이드인 효과 추가

[#228]

* refactor: 피드백 반영

[#228]

* refactor: 피드백 반영

[#228]

* refactor: 안쓰는 코드 제거

[#228]

feb-dain committed:
fix: 플렉스 박스 `align-items`가 동작하지 않는 에러를 개선한다 (#237)

[#236]

gabrielyoon7 committed:
feat: Alert 컴포넌트를 구현한다 (#241)

* feat: Alert 컴포넌트 구현

* refactor: props 이름 변경

* test: description 수정

* test: description 수정

* test: description 수정

feb-dain committed:
chore: 스토리북에 msw을 적용한다 (#244)

* chore: msw-storybook-addon 설치

[#243]

* chore: 스토리북에 msw 초기 세팅

[#243]

gabrielyoon7 committed:
feat: 충전소 상세 정보 조회기능을 개선하고, 누적 고장 신고 확인 기능을 구현한다 (#245)

* feat: 충전소 상세정보 조회 시 누적 신고 횟수가 보이는 기능 구현

- reportCount 프로퍼티 추가

* refactor: 충전소 상세 정보 위치 개선

* refactor: 폴더 구조 개선

* fix: 오탈자 수정

* fix: 오탈자 수정

* refactor: 파일 구조 개선

* design: 충전소 기본 정보 디자인 수정

* design: Alert 적용

* feat: 충전소 디자인 컴포넌트 및 업데이트 시간 차이 계산 기능 구현

* test: ChargerCard 추가

* feat: 충전기 상태 코드 설명 객체 추가

* refactor: 충전소 상태 객체 정의 및 타입 설정

* design: 충전기 상태 카드가 2열로 표시되도록 개선

* refactor: latestUpdateTime 타입 수정

* design: 충전기 카드 디자인 개선

* test: 날짜 버그 수정

* fix: 타입 에러 수정

* refactor: mock데이터에서 고장 신고가 좀 더 많이 나오도록 개선

* design: 디자인 일부 개선

- 간격 조절

* refactor: 불필요한 코드 제거

* refactor: 디자인 개선

* test: ChargerCard에 auto docs 추가

* test: ChargerCard, DetailedStation에 auto docs 추가

* refactor: 필요없는 as 제거

[#238]

* Update frontend/src/components/ui/DetailedStationInfo/ChargerCard.tsx

Co-authored-by: Dain Lee <108778921+feb-dain@users.noreply.github.com>

* refactor: prop 사용 코드 개선

* refactor: 주석 제거

---------

Co-authored-by: Dain Lee <feb.dain@gmail.com>
Co-authored-by: Dain Lee <108778921+feb-dain@users.noreply.github.com>

gabrielyoon7 committed:
refactor: 지도 api key 검사기능을 개선한다 (#250)

feb-dain committed:
feat: 검색어 입력시 쿼리를 보내는 기능을 구현한다 (#251)

* refactor: 스타일 이름 명확하게 변경

[#225]

* feat: 로고 아이콘 컴포넌트 구현

[#225]

* refactor: 로고 아이콘 변경

[#225]

* refactor: 급속, 완속 상수화

[#225]

* rename: getObjectKeysType → getObjectKeysWithType으로 파일 이름 변경

[#225]

* feat: `Object.fromEntries` 타입 추론 가능하게 하는 유틸 함수 구현

[#225]

* rename: getObjectKeysWithType → getTypedObjectKeys 파일 이름 변경

[#225]

* refactor: `getTypedObjectFromEntries`활용

[#225]

* comment: jsdocs 설명 추가

[#225]

* chore: mocks 폴더 절대 경로 추가

[#225]

* refactor: `selectedStationId`가 존재할 때만 쿼리를 실행시켜 불필요한 네트워크 요청 방지

[#225]

* chore: QueryClient 설정

[#225]

* feat: mock 데이터 생성, msw 연동

[#225]

* feat: msw 연동, 검색어 입력시 해당 충전소가 검색 결과에 나오는 기능 구현

[#225]

kyw0716 committed:
refactor: 누적 혼잡도 시각화 컴포넌트의 재사용성을 높인다 (#232)

* feat: Graph 컴포넌트 추가

- 리액트 컴파운드 패턴을 사용하는 컴포넌트

* refactor: Graph 컴포넌트를 활용해 StatisticsGraph 컴포넌트 구현

* test: StatisticsGraph 컴포넌트 스토리북 추가

* feat: 가로 정렬 그래프 구현

* refactor: useSyncExternalStore 사용하던 부분 Context api로 변경

- useSyncExternalStore 사용시 독립적인 컴포넌트들이 상태 공유를 하는 문제가 있어 Context api를 사용하는 방식으로 수정함.

* refactor: FlexBox 속성 활용

* refactor: Graph.BarContainer 컴포넌트 너비, 높이 고정

* refactor: 요일 선택 버튼 내부 Text variant 변경

* refactor: 스타일 가이드 props 활용

* refactor: render props 리턴 타입 ReactNode로 수정

* refactor: if문에 중괄호 추가

* fix: mock data 오류 수정

be-student commented on backend/src/main/java/com/carffeine/carffeine/service/auth/AuthService.java:
```diff
@@ -0,0 +1,25 @@
+public class AuthService {
+
+    private final MemberRepository memberRepository;
+    private final OAuthRequester restTemplateOauthRequester;
```
이름이 restTemplateOauthRequester 라는 부분 대신 oauthRequester 로 바뀌면 더 좋을 것 같아요

be-student commented on backend/src/main/java/com/carffeine/carffeine/service/auth/dto/OAuthTokenResponse.java:
```diff
@@ -0,0 +1,12 @@
+import com.fasterxml.jackson.annotation.JsonProperty;
+
+public record OAuthTokenResponse(
+        @JsonProperty("access_token")
```
자동으로 바뀌지 않나보네요 신기하네요 👍

be-student reviewed (approved):
고생많으셨습니다 👍

sosow0212 committed:
feat: 서버에서 충전소 정보들을 필터링 하는 기능을 만든다 (#218)

* feat: 필터링 기능 추가

* test: 필터링 기능 테스트 및 기존 코드 리팩토링 진행

* feat: List<Station>을 관리하는 객체 생성

* test: List<Station>을 관리하는 객체 테스트

* refactor: getValue()에서 다른 부분과 겹치지 않기 위해서 따로 getter 재정의

* feat: 인덱스 적용 및 효율적인 연산을 위해서 fetch join 쿼리에서 BETWEEN 연산자 제거 및 범위 쿼리 사용

* refactor: 불필요한 인덱스 제거

* refactor: 코드 컨벤션 정리 및 일부 리팩토링 진행

* refactor: 에러 방지를 위해 fixture의 날짜를 수동으로 설정

* refactor: 네이밍 및 메서드 구조 변경

* refactor: 네이밍 및 메서드 구조 변경

* refactor: 테스트 통합화

* refactor: 데이터를 가져올 때 INNER JOIN으로 가져와서 null인 값들은 제외하도록 변경

* refactor: 메서드 분리

* refactor: 값을 복사하고 반환하도록 변경

drunkenhw committed:
feat: 충전소의 잘못된 정보를 제보하는 기능을 구현한다  (#235)

* feat: 충전소의 잘못된 정보 도메인 추가

[#182]

* feat: 충전소의 잘못된 정보를 저장하는 기능 추가

[#182]

* feat: 충전소의 잘못된 정보를 제보하는 기능 추가

[#182]

* feat: 충전소의 잘못된 정보를 제보하는 API 추가

[#182]

* feat: id 생성 전략 추가

* feat: 제약 조건 추가

drunkenhw committed:
feat: 충전소 상세 조회 시 신고 횟수도 반환하도록 추가 (#204)

[#179]

drunkenhw committed:
refactor: 혼잡도를 저장하는 기능을 개선한다 (#249)

* refactor: 혼잡도 저장 로직 변경

* refactor: 충전기 상태 저장 로직 변경

* fix: 잘못된 로직 변경

* refactor: 새로운 충전소를 추가하는 기능에서 상태를 업데이트 하는 로직 제거

drunkenhw committed:
fix: 테스트 실패하는 버그 수정 (#257)

sosow0212 reviewed (approved):
수고하셨습니다👍

gabrielyoon7 committed:
feat: 충전기 고장 신고 기능을 구현한다 (#258)

* feat: 충전소 및 충전기 제보 버튼 추가

* chore: target를 ESNext으로 변경

- 기존 es5

* feat: 충전기 신고 msw 로직 구현

* feat: 충전기 신고 확정을 확인하는 모달 구현

* feat: 충전기 신고 확정을 확인하는 기능 구현

* feat: 충전기 신고 여부를 조회하는 msw 추가

* chore: compilerOptions target을 ES2021으로 수정

* chore: typescript compiler target을 ESNext에서 ES2021으로 변경

* refactor: 고장 신고 기능을 tanstack-query에서 하도록 기능 개선

* feat: tanstack query로 기존 신고 여부 조회하는 기능 구현

* fix: msw 오류 수정

* chore: msw 1.2.3 버전으로 업데이트
- mswjs/msw#1640

* feat: react query를 활용하여 신고 여부 확인하는 기능 구현

* refactor: 변수 명을 좀 더 명확하게 개선

* refactor: 코드 분리 및 디자인 개선

* refactor: 코드 분리

* refactor: 피드백 반영
- api 명세에 맞춰서 fetch함수 수정

* refactor: 피드백 반영
- 불필요한 변수 및 주석 제거

* refactor: DEFAULT_TOKEN 상수화

* refactor: 충전기 고장 신고 문구 수정

kyw0716 committed:
refactor: 충전소 리스트 UI, 필터링 UI를 개선한다 (#262)

* refactor: text-align 디폴트 속성 수정

- left → inherit

* design: 필터링 컴포넌트 너비 수정

* design: 충전소 리스트 컴포넌트 스타일 추가

* refactor: 불필요한 isFetching 제거

- isFetching에 따라 다른 렌더링을 진행했을 때 사용자가 깜빡이는 화면을 계속해서 봐야하는 문제점이 있어 이를 개선하고자 isFetching에 의한 분기처리를 제거했다.

* design: StationList 컴포넌트 width, height 속성 수정

- 컨테이너의 크기가 지정되면 해당 컨테이너의 너비, 높이에 꽉 차도록 스타일을 수정했다.
- 아직 StationCard 컴포넌트는 너비가 지정되어 있으므로 이는 유의 바람

* refactor: 기존 스타일 재사용

* fix: 명세 변경에 따른 타입 수정

be-student committed:
feat: 검색을 하는 기능을 구현한다 (#254)

* feat: gitignore 에 mysql 을 추가한다

* feat: docker-compose 와 같은 port 를 바라보도록 application.yml 수정한다

* feat: 이름과 주소를 기준으로 검색하는 기능을 추가한다

* feat: 응답을 하는 dto 를 작성한다

* feat: station 을 검색하는 기능을 추가한다

* feat: 컨트롤러에서 API 응답을 보내는 기능을 추가한다

* feat: speed 를 계산하는 기능을 추가한다

* fix: stationName 검색 부분을 더 직관적으로 처리한다

sosow0212 committed:
fix: 배포시 ec2 db접근 안되는 문제 해결 (username, pw변경으로 인해) (#269)

drunkenhw committed:
API 명세와 다른 부분을 수정한다 (#271)

* fix: api 명세와 다른 부분 수정

* fix: rest docs 수정

drunkenhw committed:
fix: heap memory 설정 (#270)

kiarakim committed:
feat: 토큰 생성 및 검증 기능을 구현한다 (#216)

* chore: spring security 의존성 추가

* chore: Json Web Token 의존성 추가

* feat: Token 생성 기능 구현

* feat: Token 검증 기능 구현

* chore: oAuth 의존성 추가

* feat: Member 엔티티 및 테스트 코드 작성

* feat: 인증 서비스 작성

* feat: id로 토큰 생성 기능 구현

* test: 토큰 만료 검증 테스트 작성

* comment: 주석 및 로그 제거

* feat: 토큰의 id로 DB Member 반환 기능 구현

* feat: Controller 작성

* refactor: 토큰 생성 및 검증 기능 수정

* refactor: User 도메인에 id 추가

* feat: Member의 role 제거

* refactor: 토큰 생성 방식 변경

* refactor: 토큰 검증 방식 변경

* refactor: Jwt 객체 사용

* test: 빈 상속 클래스 생성

* chore: oauth 의존성 라이브러리 제거

* refactor: 사용하지 않는 파라미터 제거

* refactor: 패키지 명 변경

* refactor: 클래스 네이밍 변경 User -> Member

* refactor: 컨벤션 적용

* refactor: 메서드 분리

* feat: 플렉스 박스 스타일 컴포넌트를 구현한다 (#173)

* test: 필요없는 파일 및 코드 삭제

[#114]

* test: 타입 중복 제거

- 재사용되는 타입은 전역으로 분리
- 컴포넌트의 Props export하여 스토리북에서 재사용

[#114]

* feat: FlexBox 컴포넌트 구현

[#114]

* design: GlobalStyle에서 `font` 적용하는 선택자 변경

이유
- font가 적용 안 되는 상황 방지
- 스토리북 font도 Noto Sans KR로 변경

[#114]

* chore: storybook 접근성 검사 addon 설치

[#114]

* refactor: 공통으로 사용되는 스타일 전역으로 분리

[#114]

* fix: 공통 style 전역으로 분리하면서 생기는 에러 개선

절대 경로 설정

[#114]

* refactor: 타입 이름 명확하게 변경

[#114]

* refactor: `justify content`와 `align items` 속성 따로 적용할 수 있도록 변경

[#114]

* test: row gap과 column gap 설명 보강

[#114]

* refactor: px → rem 단위로 변경

[#114]

* refactor: FlexBox에 width, height 속성을 추가한다 (#178)

* refactor: FlexBox에 width, height 속성 추가

[#177]

* refactor: width/height를 적용하지 않았을 때, 어떤 값이 적용되는지 코드만 봐도 알 수 있도록 개선

[#177]

* fix: 스토리북 svg 경로 에러를 해결한다 (#184)

* refactor: px → rem으로 변경

[#175]

* fix: `@assets` 절대 경로 스토리북에 추가

[#183]

* chore: @svgr/webpack 설치

[#183]

* fix: storybook svg 경로 에러 개선

[#183]

* fix: @style 경로 참조 에러 관련 웹팩 설정 추가 (#190)

[#189]

* feat: 충전소 정보 서버 필터링 api를 mocking한다 (#188)

* feat: 서버 필터링 api mocking

[#169]

* refactor: 서버 필터링 부분 코드 가독성 개선

[#169]

* refactor: ,를 통해 여러 query param을 전달할 수 있도록 하는 유틸함수 추가

- URLSearchParams는 ,를 사용할 수 없어 새로운 유틸함수 추가함

[#169]

* feat: 간단한 예시 코드 추가

[#169]

* refactor: 실제 반환 데이터를 기반으로 mock data 생성하도록 수정

[#169]

* refactor: 충전기 타입, 회사명 상수화

[#169]

* feat: 충전소 세부정보에 상수화 된 충전기 타입 적용

[#169]

* refactor: 불필요한 console.log 삭제

[#169]

* refactor: 미리 상수화 해둔 객체를 활용해 mock data를 생성하도록 수정

[#169]

* refactor: url 변환 함수 코드 개선

[#169]

* fix: 잘못된 상수 수정

[#169]

* feat: 네비게이터 컴포넌트를 구현한다 (#191)

* feat: Navigator 컴포넌트 추가

[#185]

* design: StationList 컴포넌트 위치 수정

[#185]

* feat: 지도에 Navigator 컴포넌트 추가

[#185]

* feat: Navigator에 아이콘 추가

[#185]

* design: cursor pointer 속성 추가

[#185]

* refactor: 로고 변경

[#185]

* refactor: px → rem

[#185]

* feat: 실시간 필터링 옵션에 외부인 출입 제한 여부에 대한 선택 옵션 추가 (#193)

[#186]

* refactor: Text variant type 지정 (#194)

[#192]

* feat: 충전소를 고장 신고하는 기능을 구현한다 (#196)

* feat: 고장 신고 클래스 추가

[#180]

* feat: 고장 신고 저장하는 기능 추가

[#180]

* feat: 고장 신고 기능 추가

[#180]

* feat: 고장 신고 api 추가

[#180]

* feat: 이미 신고한 충전소 검증 기능 추가

[#180]

* feat: List와 ListItem 컴포넌트를 구현한다 (#197)

* feat: List 컴포넌트 추가

[#172]

* feat: ListItem 컴포넌트 추가

[#172]

* fix: 오류로 인해 사라진 ListItem 컴포넌트 복구

[#172]

* feat: List Storybook 및 ListItem readme 추가

* feat: List에 디자인 옵션 추가

[#172]

* feat: ListItem에 clickable과 divider 추가

[#172]

* test: Menu 예제 추가

[#172]

* fix: Text 스토리타입 타입 오류 수정

[#172]

* test: List로 제어할 수 있는 예제 추가

[#172]

* feat: CSSProp 기능 추가

[#172]

* test: ListItem 스토리북 예제 구현

[#172]

* refactor: Text mb가 0일 때 아무것도 하지 않도록 개선

[#172]

* feat: 사용자 식별이 가능하도록 토큰을 생성하는 기능 구현 (#199)

* feat: 최초 접속 시 랜덤한 8자리 정수의 토큰을 발행하여 로컬 스토리지에 저장하는 기능 구현

- 단, 이미 발행된 토큰이 발견되는 경우에는 재발급 하지 않음

* refactor: 불필요한 콘솔 제거

* refactor: 스토리지 키 위치 수정

[#198]

* refactor: 절대 경로 확장성 있게 수정한다 (#205)

[#203]

* chore: `@heroicons/react`를 설치한다 (#202)

[#200]

* chore: `@heroicons/react`를 설치한다 (#202)

[#200]

* feat: develop mode에서 msw와 로컬 서버를 동적으로 선택하는 기능을 구현한다 (#215)

* refactor: msw를 실행하는 함수 분리

[#210]

* feat: msw를 키고 끄는 버튼 추가

[#210]

* feat: msw 컨트롤 버튼이 development 모드에서만 보이도록 수정

[#210]

* docs: 프론트엔드 설치 및 실행 방법 설명 추가

[#210]

* refactor: isMswMode로 상태명 변경

[#210]

* feat: 서버 필터링 옵션을 선택하는 UI를 구현한다 (#207)

* refactor: FilterButtonList → StationFilterButtonList로 네이밍 변경

[#187]

* feat: 서버 필터링 기능 선택 UI 추가

[#187]

* feat: Navigator 컴포넌트에서 ServerStationFilters 컴포넌트 열고 닫는 기능 구현

[#187]

* refactor: StationFilterButtonList → ClientStationFilters로 네이밍 변경

[#187]

* refactor: 충전 속도 타입 추가

[#187]

* feat: 지도에 ServerStationFilter 컴포넌트 추가

[#187]

* design: 필터링 컴포넌트 우측 border 추가

[#187]

* feat: 서버 필터링 선택 UI 추가

[#187]

* refactor: FlexBox 컴포넌트 props 추가 → alignItems

[#187]

* fix: 잘못 들어간 타입 수정

[#187]

* style: props 넘길 때 '' → {''}로 수정

[#187]

* feat: 서버 필터링 옵션 적용하기 기능 구현

[#187]

* refactor: map을 참조하는 방식 수정

- 기존에 map을 인자로 넘겨 받아 수행하던 작업들 store에서 직접 참조하는 방식으로 변경

[#187]

* refactor: 필터링 버튼 사이의 간격 추가

[#187]

* design: 아이콘 크기 수정

[#187]

* refactor: 색상 표기 방식 변경 black → #000

[#187]

* refactor: 절대 경로 수정

[#187]

* design: 우아한 다크모드 삭제

[#187]

* feat: Box 컴포넌트를 구현한다 (#219)

* feat: Box 컴포넌트 추가

[#217]

* feat: border 추가

[#217]

* test: border 예제 추가

[#217]

* feat: p 기능 구현

[#217]

* feat: p 기능 완성

[#217]

* feat: m 기능 완성

[#217]

* feat: 높이, 너비 기능 완성

[#217]

* test: 높이, 너비 테스트

[#217]

* feat: bgColor, color 구현

[#217]

* test: bgColor, color 스토리북 작성

[#217]

* feat: position 관련 기능 추가

[#217]

* test: position 관련 기능 예제 추가

[#217]

* test: 부족한 description 추가

[#217]

* test: 부족한 description 추가

[#217]

* refactor: 리뷰 반영

[#217]

* feat: 고장 신고 여부를 확인하는 기능을 구현한다  (#201)

* feat: 고장 신고를 회원과 충전소 기준으로 찾는 기능 추가

[#181]

* feat: 고장 신고를 회원과 충전소 기준으로 찾는 API 추가

[#181]

* feat: 스토리북 배포를 자동화한다 (#206)

* feat: TextField를 구현한다 (#222)

* fix: 잘못된 속성 제거

[#220]

* feat: 기본 Input 컴포넌트 작성

[#220]

* feat: TextField로 변경 및 디자인 추가

[#220]

* refactor: 불필요한 컴포넌트 제거

[#220]

* feat: placeHolder 기능 추가 및 css 프롭 분리

[#220]

* feat: value, onChange 추가

[#220]

* feat: helperText 구현

[#220]

* test: description 개선

[#220]

* feat: width 기능 구현

[#220]

* refactor: 피드백 반영

[#220]

* refactor: 함수 순서 변경

[#220]

* refactor: 피드백 반영

[#220]

* feat: 검색창 컴포넌트를 구현한다 (#221)

* refactor: px → rem으로 변경

[#175]

* feat: StationSearch 컴포넌트 구현

[#175]

* refactor: getSize 함수 전역으로 관리

[#175]

* refactor: Button에 width, height 속성 추가

[#175]

* refactor: ListItem에 필요없는 속성 제거하고 NoLastDivider 속성 추가

[#175]

* refactor: gitSize 전역으로 관리

[#175]

* refactor: Text 타입 깔끔하게 분리

[#175]

* refactor: 검색 컴포넌트 구현

[#175]

* refactor: 검색바 분리 및 속성 추가

[#175]

* refactor: Text 스타일 컴포넌트에 새로운 속성 추가

[#175]

* refactor: 스타일 아래로 이동

[#175]

* refactor: 필요없는 코드 제거

[#175]

* feat: 검색 결과 리스트 컴포넌트 구현

[#175]

* feat: 충전소 검색창 컴포넌트 구현

[#175]

* feat: 누적 혼잡도 통계를 시각화 해주는 컴포넌트를 개발한다 (#223)

* refactor: 불필요한 import 삭제

* refactor: 허스키 삭제

* refactor: 필터 적용하기 버튼 sticky 적용

* feat: 수치에 따라 길이가 변하는 그래프 바 구현

* feat: 혼잡도 통계 그래프 UI 구현

- 이전에 구현한 그래프 바 이용
- 헤더에 요일 버튼 UI 구현
- 그래프 바에 시간 정보 추가

* feat: 누적 혼잡도 ratio가 -1일 경우 회색으로 표시하는 기능 구현

* refactor: 구조분해 할당 적용

* feat: 급속/완속, 요일 변경 기능 추가

* feat: 선택된 요일 표시 기능 추가

* refactor: 급속/완속 토글 메서드 코드 개선

- if 분기처리 → 삼항연산자

* feat: pr시 comment에 스토리북 배포 링크를 자동으로 달아주는 기능을 구현한다 (#227)

[#226]

* refactor: Member객체 대신 id를 반환하도록 변경

* refactor: 어노테이션 선언 위치 변경

* refactor: 빈줄 컨벤션 적용

* test: mock bean 주입 클래스 상속

* refactor: 사용하지 않는 api 및 Controller 제거

* refactor: 클래스 네이밍 변경

* refactor: secret을 한 곳에서 관리하도록 수정

* refactor: 개행 추가

* refactor: 사용하지 않는 인스턴스 제거

* refactor: JwtProvider의 기능을 Jwt에 통합

* refactor: 부분 문자열 추출 범위 수정

* refactor: 토큰 만료 기간 암호화

* refactor: 토큰 만료 기간 암호화

* refactor: Optional 처리 및 커스텀 예외 사용

* refactor: 컴포넌트 클래스 의존성 주입으로 변경

* refactor: Key 한번만 생성하도록 변경

* refactor: 에러처리

---------

Co-authored-by: Dain Lee <108778921+feb-dain@users.noreply.github.com>
Co-authored-by: YoungWoo Kim <77326660+kyw0716@users.noreply.github.com>
Co-authored-by: Gabriel Ju Hyun, Yoon <gabrielyoon7@gmail.com>
Co-authored-by: 한우석 <106640954+drunkenhw@users.noreply.github.com>

sosow0212 committed:
feat: Spring actuator 를 설정한다 (#261)

* chore: Actuator 종속성 및 Prometheus 에서 매트릭 수집을 위한 종속성 추가

* feat: Actuator 포트 설정 및 필수 엔드 포인트 설정

* refactor: 주석 제거 및 포트 제거

drunkenhw committed:
refactor: 패키지 구조를 변경한다 (#273)

* refactor: 패키지 구조 변경

* refactor: 패키지 구조 변경

sosow0212 committed:
refactor: 환경변수로 데이터베이스 정보 받도록 변경 (#276)

be-student committed:
feat: cors 와 timezone 을 설정한다 (#279)

* feat: timezone 설정을 추가

* feat: cors 설정을 해제한다

gabrielyoon7 committed:
feat: 충전소 정보 수정 제안 기능을 구현한다 (#281)

* refactor: 충전소 정보 수정 제안 버튼 문구 수정

* feat: 충전소 정보 제안 모달 기본 틀 구현

* fix: 충전소 정보 상세 페이지가 스크롤 되지 않는 문제 개선

* refactor: StationInformation 분리

* feat: StationInformation가  StationReportConfirmation 모달에도 출력되는 기능 구현

[#259]

* feat: TextField에 fullWidth 기능 추가

* test: TextField fullWidth

[#259]

* feat: 충전소 정보 수정 제안 폼 추가

[#259]

* refactor: form을 TextField와 input에 연동

[#259]

* refactor: 모든 TextField가 form과 연동되도록 개선

[#259]

* refactor: 충전기 불일치 신고 삭제

[#259]

* feat: 원본 충전소 정보와, 수정된 충전소 정보를 비교하여 수정된 부분만 객체로 추출해내는 기능 구현

[#259]

* feat: 수정된 객체의 프로퍼티만 추출하여 배열로 만드는 기능 구현

[#259]

* feat: fetchReportStation 구현

[#259]

* feat: msw에서 사용할 /stations/:stationId/misinformation-reports 구현

[#259]

* refactor: 피드백 반영

[#259]

feb-dain committed:
feat: 충전소 검색시 해당 충전소 마커로 지도 이동 및 충전소 정보 렌더링하는 기능을 구현한다 (#283)

* design: 디자인 수정

[#252]

* design: 닫기 버튼 디자인 변경 및 버튼에 새로운 디자인 추가

[#252]

* refactor: 커스텀 아이콘 분리

[#252]

* feat: 검색창이 focus 됐을 때만 검색 결과가 보이는 기능 구현

+ 검색 결과창 디자인 변경

[#252]

* feat: 검색할 때마다 검색 결과가 업데이트 되는 기능 구현

[#252]

* feat: 검색하면 검색어와 가장 흡사한 이름 가진 충전소로 이동하는 기능 구현

[#252]

* feat: 관련 충전소로 이동하는 기능 구현

- 검색어 입력 → 가장 근접한 이름 가진 충전소로 이동
- 검색 후 검색 결과에 나온 충전소 클릭 → 해당 충전소로 이동

[#252]

* refactor: 변경된 api에 맞춰 타입 수정

[#252]

* refactor: 타입 재사용

[#252]

* refactor: 코드 간결화

[#252]

* fix: StationSearchBar 스토리북 에러 개선

[#252]

* feat: 검색창, 검색 결과가 아닌 다른 곳을 누르면 검색 결과가 안 보이는 기능 구현

[#252]

* refactor: 검색 결과가 없을 때 검색 결과 없음 문구 나오도록 수정

[#252]

kyw0716 committed:
feat: 아코디언 컴포넌트를 개발한다 (#280)

* feat: 아코디언 컴포넌트 구현

* feat: Navigator컴포넌트 Accordion 컴포넌트에 병합

* refactor: 지도 UI에 아코디언 컴포넌트 적용

* refactor: 충전소 상세 정보 UI에 아코디언 컴포넌트 적용

* design: 아이콘 변경

[#266]

---------

Co-authored-by: Dain Lee <feb.dain@gmail.com>
Co-authored-by: Dain Lee <108778921+feb-dain@users.noreply.github.com>

feb-dain committed:
fix: 충돌 해결 (#285)

[#284]

gabrielyoon7 committed:
fix: 지도 이동 시 두번 씩 fetching을 시도하려는 문제 수정 (#286)

feb-dain committed:
fix: 데이터 받아오는 부분 api 명세대로 변경한다 (#291)

* fix: 데이터 받아오는 api 명세대로 변경

[#290]

* refactor: 타입명 명확하게 변경

[#290]

drunkenhw committed:
메서드 잘못 호출한 부분을 수정한다 (#296)

* fix: 메서드 잘못 호출한 부분 수정

* fix: 테스트 깨지는 부분 수정

[#295]

be-student committed:
feat: 기본 페이지를 1로 선택해서 검색 (#289)

be-student committed:
fix: 커넥션 풀 사이즈를 22개로 늘린다 (#287)

drunkenhw committed:
prod, dev 서버를 나눠 배포한다 (#294)

* chore: prod yml 추가

* chore: dev action 추가

* chore: web hook url 추가

drunkenhw committed:
chore: prod 추가 (#298)

[#297]

drunkenhw committed:
chore: prod dev profile 분리 (#300)

[#299]

drunkenhw committed:
chore: prod dev profile 분리 (#303)

[#302]

gabrielyoon7 committed:
refactor: 실제 url에 연결하고 development 모드에서 서버를 유동적으로 선택할 수 있도록 개선한다 (#304)

* refactor: DevelopmentServerControlButton 추가

[#293]

* feat: 서버를 전역 상태로 선택하는 기능 구현

[#293]

* refactor: 기존의 서버 선택기를 제거하고 새로운 선택기로 대체

[#293]

* feat: msw on 상태가 되면 자동으로 localhost로 연결되도록 개선

[#293]

* refactor: servers를 SERVERS으로 이름 변경

[#293]

* refactor: import 수정

[#293]

* refactor: 변수명 개선

[#293]

* refactor: useUpdateStations에 기본 Fn 부여

[#293]

kiarakim committed:
refactor: prod와 dev의 DB 접속 주소를 분리한다 (#306)

gabrielyoon7 committed:
fix: useUpdateStations의 기본 Fn 버그 수정 (#308)

[#307]

feb-dain committed:
refactor: 충전소 검색창에 임의로 디자인된 주변 충전소 목록을 추가한다 (#313)

* design: 아코디언 버튼 디자인 변경

[#312]

* refactor: 검색창에 주변 충전소 목록 추가

[#312]

* design: 버튼 디자인 수정

[#312]

* refactor: 필요없는 코드 제거

[#312]

gabrielyoon7 committed:
refactor: 지도의 줌 레벨이 15이상인 경우에만 충전소 데이터를 요청하도록 개선 (#311)

Co-authored-by: kiarakim <kiarakim413@gmail.com>

kyw0716 committed:
feat: 충전소 상세 정보 컴포넌트에 누적 혼잡도 시각화 컴포넌트를 렌더링 한다 (#320)

* feat: 혼잡도 요청 api mocking

* feat: 충전소 상세 정보창에 누적 혼잡도 그래프 추가

* feat: 로딩 처리 이모지 (⌛) 추가

* refactor: 상수 사용

be-student committed:
feat: limit 을 기본값 12개로 변경하고, 클라이언트가 수정할 수 있도록 변경한다 (#310)

kyw0716 committed:
fix: 충전소 간단 정보 window에서 아코디언을 펼칠 수 없는 버그를 개선한다 (#321)

* fix: 충전소 간단 정보창에서 상세보기 클릭시 아코디언 패널이 열리도록 수정

* feat: 화면내 충전소 정보가 없다면 모든 아코디언 패널을 닫는 기능 추가

* refactor: boolean state에 ~is prefix 추가

gabrielyoon7 committed:
feat: 개선된 버튼 컴포넌트 구현한다 (#322)

* feat: ButtonNext 파일 추가

[#314]

* feat: variant 프롭 추가

[#314]

* refactor: INITIAL_ZOOM_SIZE 적용

* feat: Button variant 추가

* feat: Button color 지정

* feat: noTheme 옵션 추가

* feat: size 및 호버 효과 구현

* feat: text, outlined에도 호버 효과 구현

* feat: text, outlined hover 효과 개선

* test: disabled 예제 추가

* test: Color타입 분리

* refactor: 사이즈, 색상 호출하는 코드 개선

* refactor: 색상 예제 추가

* test: 버튼 사이즈 예제 추가

* test: disabled 디자인 개선

* feat: fullWidth 구현 및 예제 추가

* fix: 일부 프롭이 전해지지 않던 버그 수정

* feat: pill 옵션 추가

* test: pill 옵션 추가

* test: description 업데이트

feb-dain committed:
feat: 디바운스 이용해 검색 기능 최적화한다 (#323)

* design: 아코디언 버튼 디자인 변경

[#312]

* refactor: 검색창에 주변 충전소 목록 추가

[#312]

* design: 버튼 디자인 수정

[#312]

* refactor: 필요없는 코드 제거

[#312]

* refactor: 웹 접근성 개선

[#317]

* refactor: 검색창이 열린 상태를 기본값으로 설정

[#317]

* refactor: useDebounce 이용해 검색 기능 최적화

[#317]

kiarakim committed:
fix: 변경된 db 접속 주소를 적용한다 (#329)

* fix: 변경된 db 접속 주소를 적용한다

* fix: 변경된 db 접속 주소를 적용한다

feb-dain committed:
fix: 초기 충전소 정보가 없어 발생하는 런타임 에러 개선한다 (#330)

* fix: 초기 충전소 정보가 없어 발생하는 런타임 에러 개선

Co-authored-by: kyw0716 <kyw0716@naver.com>

[#326]

* comment: 필요없는 주석 제거

[#326]

drunkenhw committed:
fix: cors origin 설정 변경 (#331)

[#318]

drunkenhw committed:
feat: 스케줄링을 profile에 따라 분리한다 (#334)

* feat: 스케줄링을 production 서버에만 사용하도록 변경

[#333]

* feat: 처음 scrap을 production 서버에만 사용하도록 변경

[#333]

* feat: 처음 scrap을 production 서버에만 사용하도록 변경

[#333]

* feat: scheduling 분리

[#333]

gabrielyoon7 committed:
refactor: 기존 Button 컴포넌트를 ButtonNext로 교체한다 (#336)

* refactor: FilterOption의 Button을 ButtonNext로 교체

* refactor: CongestionStatistics의 Button을 ButtonNext로 변경

* refactor: MswControlButton의 Button을 ButtonNext로 변경

* refactor: msw 버튼 디자인 개선

* refactor: MapController에 ButtonNext 적용

* refactor: CircleDaySelectButton에 ButtonNext 적용

[#328]

* refactor: ApiKeyChecker에 ButtonNext 적용

[#328]

* refactor: Confirmation에 ButtonNext 적용

[#328]

* refactor: ClientStationFilters에 ButtonNext 적용

[#328]

drunkenhw committed:
feat: storybook 배포 스크립트를 수정한다  (#337)

* chore: 도커 파일 배포 기본 설정

[#315]

* fix: 오탈자 수정

[#315]

* fix: 도커 파일 수정

[#315]

* fix: husky 스크립트 삭제

[#315]

* fix: docker image 이름 수정

[#315]

* chore: prod dev dockerfile 분리

[#315]

* chore: development build script 추가

[#315]

* chore: 도커 파일 배포 기본 설정

[#315]

* fix: 오탈자 수정

[#315]

* fix: 도커 파일 수정

[#315]

* fix: husky 스크립트 삭제

[#315]

* fix: docker image 이름 수정

[#315]

* chore: prod dev dockerfile 분리

[#315]

* chore: development build script 추가

[#315]

* fix: 배포 스크립트 경로 수정

[#315]

---------

Co-authored-by: gabrielyoon7 <gabrielyoon7@gmail.com>

be-student committed:
fix: 혼잡도응답 dto 순서 수정 (#341)

feb-dain committed:
design: 충전소 마커 클릭시 나오는 간단 정보 모달에 임의 디자인을 넣는다 (#343)

* feat: 충전소 간단 정보 모달 임의 디자인 부여

[#338]

* refactor: 스타일 컴포넌트에 html 속성 부여

[#338]

* refactor: 검색 결과 웹 접근성 개선

[#338]

* feat: 검색시 가장 이름이 근접한 충전소 상세 정보 보여주는 기능 구현

[#338]

kyw0716 committed:
fix: 키값 참조가 잘못 되어 발생하던 런타임 에러 수정 (#342)

gabrielyoon7 committed:
refactor: 기존의 google.maps.Marker를 google.maps.marker.AdvancedMarkerElement으로 교체한다 (#344)

* refactor: 최신 지도 적용 및 지도가 idle인 경우에만 쿼리를 요청하도록 개선

[#167]

* refactor: AdvancedMarker 적용

[#167]

* refactor: msw 모드를 전역 상태로 관리 및 AdvancedMarkerElement에 마커 렌더링 복구

[#167]

* refactor: 오탈자 수정

[#167]

* refactor: 오탈자 수정

[#167]

gabrielyoon7 committed:
refactor: api key 검사기 제거한다 (#346)

[#345]

gabrielyoon7 committed:
refactor: .env으로 google maps api key을 설정한다 (#348)

* refactor: .env으로 google maps api key 설정

[#347]

* refactor: 불필요한 스크립트문 제거

[#347]

gabrielyoon7 committed:
fix: webpack production mode build 오류를 수정한다 (#350)

* refactor: .env으로 google maps api key 설정

[#347]

* refactor: 불필요한 스크립트문 제거

[#347]

* fix: webpack fallback 추가

[#349]

* fix: 배포 스크립트 수정

[#349]

---------

Co-authored-by: drunkenhw <drunkenhw@gmail.com>

kyw0716 committed:
refactor: 좌측 메뉴에 의해 마커가 가려지는 문제를 해결한다 (#353)

* feat: 메뉴 펼쳐짐 정도에 따라 달라지는 메뉴 너비 구하기

[#332]

* feat: 구한 메뉴 너비와 현재 브라우저의 너비의 비율 구하기

[#332]

* feat: 구한 비율 만큼의 델타 값 구하기

[#332]

* feat: 계산된 델타값 반환하는 커스텀 훅 추가

[#332]

* refactor: 마커 렌더링 관련 로직 훅 분리

[#332]

* refactor: 간단정보창을 여는 로직 훅 분리

[#332]

* feat: 네비게이션바에 의해 가려진 영역 만큼을 지도 내 델타 값으로 치환해서 중심 좌표를 이동하는 기능 구현

[#332]

* refactor: 클라이언트 필터링 옵션 수정 외부인 출입 제한 → 공공 충전소

[#332]

* refactor: 불필요한 import 삭제

[#332]

drunkenhw committed:
refactor: profile 에 따라 스케줄링 로직 변경 (#355)

[#354]

kyw0716 committed:
fix: 상세정보가 없을 때도 상세정보창이 남아있는 오류 수정 (#359)

* fix: 상세정보가 없을 때도 상세정보창이 남아있는 오류 수정

[#356]

* refactor: 사용하지 않는 import 삭제

[#356]

gabrielyoon7 committed:
refactor: 폴더 구조와 파일 이름을 개선한다 (#362)

* refactor: hooks 폴더 분리

[#360]

* rename: useSelectedStation을 useStationDetails

[#360]

* refactor: @constants/styles.ts 에서 사용하던 함수를 style/index.ts로 이동

[#360]

* refactor: 상수 분리

[#360]

* refactor: 상수 분리

[#360]

* refactor: 상수 분리

[#360]

* refactor: 상수 분리

[#360]

* refactor: 상수 분리

[#360]

* refactor: 상수 분리

[#360]

* refactor: 타입 분리

[#360]

* refactor: 타입 분리

[#360]

* refactor: 타입 분리

[#360]

Co-authored-by: kyw0716 <kyw0716@naver.com>
Co-authored-by: Dain Lee <feb.dain@gmail.com>

gabrielyoon7 committed:
feat: 충전소 정보 수정 제안 시 단순 제보 기능을 구현한다 (#364)

feb-dain committed:
chore: prettier 3.0 버전으로 올리고 type 절대 경로를 추가한다 (#372)

* chore: `@trivago/prettier-plugin-sort-imports`, `prettier` 최신 버전으로 업데이트

[#371]

* chore: `@type` 절대 경로 추가

[#371]

* chore: 절대 경로로 변경

[#371]

kyw0716 committed:
refactor: 혼잡도 그래프가 신고시에 로딩 표시를 띄우는 문제 개선 (#368)

- reactQuery의 isFetching으로 되어 있던 부분을 isLoading으로 고쳐서 개선
- isLoading은 캐시된 데이터가 없고, 처음 로딩될 때 true이고 isFetching은 그냥 동기화 작업이 진행될 때마다 true가 되어 발생한 문제였음

[#361]

gabrielyoon7 committed:
feat: 스켈레톤을 구현한다 (#373)

* feat: Skeleton 구현

[#367]

* test: Skeleton 스토리북 작성

[#367]

* fix: 누락 파일 추가

[#367]

* refactor: 피드백 반영

[#367]

drunkenhw committed:
fix: 스케줄 동작하도록 수정 (#377)

[#376]

drunkenhw committed:
refactor: 로그인 시 jwt 발급하도록 변경

[#87]

drunkenhw committed:
Merge remote-tracking branch 'origin/feat/87' into feat/87

# Conflicts:
#	backend/src/main/resources/application.yml
#	backend/src/test/java/com/carffeine/carffeine/fake/member/FakeMemberRepository.java
#	backend/src/test/resources/application.yml

[#87]
